src/controllers/Auth.controller.ts

```diff
@@ -0,0 +1,86 @@
+import {AuthorizationAPI} from '../api/Authorization.api';
+import {router} from '../utils/router/Router';
+import {routes} from '../utils/consts';
+import {errorHandler} from '../utils/services/errorHandler';
+import {Controller} from './Controller';
+import {IUserInfo} from './User.controller';
+import {initialState} from '../utils/store/Store';
+
+export interface ISignUpData {
+    first_name?: string,
+    second_name?: string,
+    login?: string,
+    email?: string,
+    password?: string,
+    phone?: string
+}
+
+export interface ISignInData {
+    login?: string,
+    email?: string,
+}
+
+const AuthFlag = 'Auth';
+
+export class AuthController extends Controller {
+    static __instance: AuthController | null;
+
+    constructor() {
+        if (AuthController.__instance) {
+            return AuthController.__instance;
+        }
+        super();
+    }
+
+    public async signUp(data: ISignUpData) {
+        AuthorizationAPI.signUp(data)
+            .then(async (res) => {
+                console.log(res);
+                await this.getUser();
+                localStorage.setItem(AuthFlag, 'true');
+                router.go(routes.messenger);
+            })
+            .catch(errorHandler)
+    }
+
+    public async signIn(data: ISignInData) {
+        AuthorizationAPI.signIn(data)
+            .then(async (res) => {
+                console.log(res);
+                await this.getUser();
+                localStorage.setItem(AuthFlag, 'true');
+                router.go(routes.messenger);
+            })
+            .catch(errorHandler)
+    }
+
+
+    public async getUser() {
+        return AuthorizationAPI.getUser()
+            .then((user: IUserInfo) => {
+                this.store.setState({user});
+                return user;
+            })
+            .catch(errorHandler)
+    }
```

Comment:
async не нужен, для того что бы он был полезен, надо написать вот так:

```suggestion
    public async getUser() {
        try {
            const user = await AuthorizationAPI.getUser();
            this.store.setState({user});
            return user;
        } catch (e) {
            errorHandler(e);
        }
    }
```